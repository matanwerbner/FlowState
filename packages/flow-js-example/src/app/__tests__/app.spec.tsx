import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import App from '../app';

global.structuredClone = (val) => JSON.parse(JSON.stringify(val));

describe('App', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  afterEach(() => {
    cleanup();
  });

  it('should render app initially', () => {
    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
});
