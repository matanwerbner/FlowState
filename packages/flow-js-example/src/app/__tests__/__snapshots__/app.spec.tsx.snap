// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render app initially 1`] = `
<body>
  <div>
    <div
      class="appstyled__StyledApp-swyx9x-0 htrBUj"
    >
      <div
        class="appstyled__GameContainer-swyx9x-1 klwzVE"
      >
        <h1
          class="appstyled__GameHeader-swyx9x-3 jhlKvI"
        >
          Trivia game
        </h1>
        <div
          class="appstyled__RouterContainer-swyx9x-2 gIiTQc"
        >
          <div
            style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 400ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 400ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
          >
            <div
              class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
              >
                Questions category
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-controls=":r0:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  role="combobox"
                  tabindex="0"
                >
                  Choose category
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  data-testid="category-select"
                  tabindex="-1"
                  value="-1"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
